### PR TITLE
Apply NRT to all files where it can be easily applied

### PR DESCRIPTION
**Pull request by peppy**
Something I've had sitting around. Tested against osu! (see https://github.com/ppy/osu/pull/24019).

Another pass will follow, but feel that getting this one out of the way is a good first step.

- 75ed421 is an automated pass. reviewing it separately is recommended as it should be a breeze
- 3064650 are manual fixes after the automated pass to fix non-compile-time warnings (aka r#)
- d002fda is stuff rider didn't pick up on (CI script unfortunately crashes locally for me)
- BOM in files is inconsistent in our repository. this arbitrarily changes BOM. please don't focus on this. we should normalise this at a later point.

### CODE REVIEW

File: osu.Framework/IO/Stores/GlyphStore.cs [hunk 1]
```diff
@@ -118,7 +118,6 @@ protected string GetFilenameForPage(int page)
             return $@"{AssetName}_{page.ToString().PadLeft((Font.Pages.Count - 1).ToString().Length, '0')}.png";
         }
 
-        [CanBeNull]
```
Comment (bdach) [inline osu.Framework/IO/Stores/GlyphStore.cs:121]: What's happened here? `#nullable disable` is not removed from the file, so `CharacterGlyph?` return type cannot be applied - so why is this removed?

File: osu.Framework/Input/Events/FocusLostEvent.cs [hunk 2]
```diff
@@ -17,8 +14,7 @@ public class FocusLostEvent : UIEvent
         /// <summary>
         /// The <see cref="Drawable"/> that will gain focus, or <c>null</c> if nothing will gain focus.
         /// </summary>
-        [CanBeNull]
-        public readonly Drawable NextFocused;
+        public readonly Drawable? NextFocused;
```
Comment (bdach) [inline osu.Framework/Input/Events/FocusLostEvent.cs:17]: Bit weird that this is nullable, but the ctor param isn't. `FocusEvent` has the ctor param nullable, too.

File: osu.Framework/Testing/DrawFrameRecordingContainer.cs [hunk 3]
```diff
@@ -33,7 +30,7 @@ public DrawFrameRecordingContainer()
         }
 
         [BackgroundDependencyLoader(true)]
```
Comment (bdach) [inline osu.Framework/Testing/DrawFrameRecordingContainer.cs:32]:
```suggestion
        [BackgroundDependencyLoader]
```

File: osu.Framework.Benchmarks/BenchmarkTextBuilder.cs [hunk 4]
```diff
@@ -51,7 +51,7 @@ private class TestStore : ITexturedGlyphLookupStore
                 new CharacterGlyph(character, character, character, character, character, null),
                 new DummyRenderer().CreateTexture(1, 1));
 
-            public Task<ITexturedCharacterGlyph> GetAsync(string fontName, char character) => Task.Run(() => Get(fontName, character));
+            public Task<ITexturedCharacterGlyph?> GetAsync(string fontName, char character) => Task.Run<ITexturedCharacterGlyph?>(() => Get(fontName, character));
```
Comment (bdach) [inline osu.Framework.Benchmarks/BenchmarkTextBuilder.cs:54]: Nitpick but I'd probably do

```diff
diff --git a/osu.Framework.Benchmarks/BenchmarkTextBuilder.cs b/osu.Framework.Benchmarks/BenchmarkTextBuilder.cs
index fb774e296..0c83350a2 100644
--- a/osu.Framework.Benchmarks/BenchmarkTextBuilder.cs
+++ b/osu.Framework.Benchmarks/BenchmarkTextBuilder.cs
@@ -47,11 +47,11 @@ private void initialiseBuilder(bool withDifferentBaselines)
 
         private class TestStore : ITexturedGlyphLookupStore
         {
-            public ITexturedCharacterGlyph Get(string fontName, char character) => new TexturedCharacterGlyph(
+            public ITexturedCharacterGlyph? Get(string fontName, char character) => new TexturedCharacterGlyph(
                 new CharacterGlyph(character, character, character, character, character, null),
                 new DummyRenderer().CreateTexture(1, 1));
 
-            public Task<ITexturedCharacterGlyph?> GetAsync(string fontName, char character) => Task.Run<ITexturedCharacterGlyph?>(() => Get(fontName, character));
+            public Task<ITexturedCharacterGlyph?> GetAsync(string fontName, char character) => Task.Run(() => Get(fontName, character));
         }
     }
 }

```

to match the nullability annotations on the interface instead?

File: osu.Framework/Testing/TestingExtensions.cs [hunk 5]
```diff
@@ -14,7 +12,7 @@ public static class TestingExtensions
         /// <summary>
         /// Find all children recursively of a specific type. As this is expensive and dangerous, it should only be used for testing purposes.
         /// </summary>
-        public static IEnumerable<T> ChildrenOfType<T>(this Drawable drawable)
+        public static IEnumerable<T> ChildrenOfType<T>(this Drawable? drawable)
```
Comment (bdach) [inline osu.Framework/Testing/TestingExtensions.cs:15]: What's with this one? Is anyone really calling `ChildrenOfType()` on a null reference?

**Comment by bdach** [comment]
Also there are some new [build](https://github.com/ppy/osu-framework/pull/5867/files#annotation_12197382190) [warnings](https://github.com/ppy/osu-framework/pull/5867/files#annotation_12197382192) in the iOS project.

**Comment by peppy** [comment]
> Also there are some new [build](https://github.com/ppy/osu-framework/pull/5867/files#annotation_12197382190) [warnings](https://github.com/ppy/osu-framework/pull/5867/files#annotation_12197382192) in the iOS project.

Ah right. Mobile platforms were a TODO on these.

File: osu.Framework.Benchmarks/BenchmarkTextBuilder.cs (hunk 4 above)
Comment (peppy) [reply osu.Framework.Benchmarks/BenchmarkTextBuilder.cs:54]: This gives a new warning, give it a try in code and see if you can come up with a better solution.

File: osu.Framework/Testing/TestingExtensions.cs (hunk 5 above)
Comment (peppy) [reply osu.Framework/Testing/TestingExtensions.cs:15]: Might be okay. But also may have been a weird osu! test usage (I likely would not have made this change without a good reason). Will need to run tests against this before applying.

File: osu.Framework/Input/Events/FocusLostEvent.cs (hunk 2 above)
Comment (peppy) [reply osu.Framework/Input/Events/FocusLostEvent.cs:17]: Fixed this, good catch.

File: osu.Framework.Benchmarks/BenchmarkTextBuilder.cs (hunk 4 above)
Comment (bdach) [reply osu.Framework.Benchmarks/BenchmarkTextBuilder.cs:54]: I even tried applying but then missed the incredibly subtle hinting that that's a warning >_>

File: osu.Framework/Testing/TestingExtensions.cs (hunk 5 above)
Comment (bdach) [reply osu.Framework/Testing/TestingExtensions.cs:15]: If you don't recall from memory then I'm not sure I care that much for you to go check. I was mostly curious is all.

Comment (peppy) [reply osu.Framework/Testing/TestingExtensions.cs:15]: Turns out it's to ease cases like this:

2023-06-24 09 43 52@2x

There's around 30 usages osu! side which would need updating if we revert this. I think making the method nullable makes it a touch more flexible though, given its intended purpose.

**Review by bdach — APPROVED** [review]
Seems good. Going to expedite merge now to minimise chances of merge conflictery here. If it turns out something goes bad we can revert in a separate pull.
